### PR TITLE
[feat] refactor listing card component

### PR DESCRIPTION
**Pull request by jinkang-0**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

- Allow listing card to take in a general listing and display the relevant data accordingly
- Update cases page card column to have a darker background

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."

image

image

Hover
image

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Pull the changes to local device and start the app to test:
```
git fetch origin jinkang/ijp-117-refactor-the-listingcard-component
git checkout jinkang/ijp-117-refactor-the-listingcard-component
git pull origin jinkang/ijp-117-refactor-the-listingcard-component
npm run dev
```

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

Before merging, delete `listing-card-test` page

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

Depends on schema changes #70 

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz

### CODE REVIEW

File: src/types/schema.d.ts [hunk 1]
```diff
@@ -75,8 +75,8 @@ export interface LimitedCaseAssignment {
   country: string;
   deadline: string; // timestamp (deadline to submit assignment)
   research_topic: string;
-  deliverable: string; // expected deliverable
-  languages: string[];
+  deliverable: string; // (new) expected deliverable
+  languages: string[]; //
```
Comment (ccatherinetan) [inline src/types/schema.d.ts]: delete the hanging comment?

File: src/components/ListingCard/index.tsx [hunk 2]
```diff
@@ -1,85 +1,133 @@
...
+    if (listing.languages && listing.languages.length > 0) {
+      let langTag = listing.languages.slice(0, 2).join(', ');
+      if (listing.languages.length > 2)
+        langTag += ` + ${listing.languages.length - 2}`;
+      tags.push(langTag);
+    }
+
+    // case interpretation
+    if (listing.listing_type === 'CASE' && interpretation) {
+      if (listing.country) tags.push(listing.country);
```
Comment (ccatherinetan) [inline src/components/ListingCard/index.tsx]: remove the country tag for case interpretation

File: src/components/ListingCard/index.tsx [hunk 3]
```diff
@@ -1,85 +1,133 @@
...
+      tags.push('Case Interpretation');
     }
 
-    // TODO: parse hrs/month if num weeks > 3
-    if (caseData.hours_per_week) {
-      tags.push(`${caseData.hours_per_week} hrs/week`);
+    // limited case assignment
+    if (listing.listing_type === 'LCA') {
+      tags.push(listing.country);
+      tags.push(listing.deliverable);
```
Comment (ccatherinetan) [inline src/components/ListingCard/index.tsx]: remove deliverable tag

File: src/components/ListingCard/index.tsx [hunk 4]
```diff
@@ -1,85 +1,131 @@
...
+        <Styles.TagRow>
+          {cardTags.map(s => (
+            <Styles.CardTag key={s} color={COLORS.blueLight}>
+              {s}
+            </Styles.CardTag>
+          ))}
+        </Styles.TagRow>
+      )}
+
+      <Flex $gap="1rem">
```
Comment (ccatherinetan) [inline src/components/ListingCard/index.tsx]: 16px instead?

**Review by ccatherinetan — APPROVED** [review]
looks good!

File: src/components/ListingCard/index.tsx [hunk 5]
```diff
@@ -1,85 +1,131 @@
...
+    // limited case assignment
+    if (listing.listing_type === 'LCA') {
+      tags.push(listing.country);
     }
 
-    if (caseData.languages && caseData.languages.length > 0) {
-      tags.push(caseData.languages.join(', '));
+    // language support
+    if (listing.listing_type === 'DOC') {
+      tags.push(`${listing.num_pages} pages`);
```
Comment (varortz) [inline src/components/ListingCard/index.tsx]: can we support the "1 page" edge case here?

**Review by varortz — APPROVED** [review]
LOOKS GOOD! let's go ahead and remove the `listing-card-test` folder before rebasing + merging

**Pull request merged**